db/flush_job.cc

```diff
@@ -244,6 +247,16 @@ Status FlushJob::Run(LogsWithPrepTracker* prep_tracker,
         ROCKS_LOG_WARN(db_options_.info_log, "Mempurge process failed: %s\n",
                        mempurge_s.ToString().c_str());
       }
+    } else {
+      if (switched_to_mempurge) {
+        *switched_to_mempurge = 1;
+      } else {
+        // The mempurge process was successful, but no switch_to_mempurge
+        // pointer provided so no way to propagate the state of flush job.
+        ROCKS_LOG_WARN(db_options_.info_log,
```

Comment:
Just curious, what are the reasons that ```switched_to_mempurge``` is not provided

Reply:
This is a good question. The first two arguments of `FlushJob::Run`, `prep_tracker` and `file_meta`, are already optional arguments, so adding a third argument required a default value as well.
In practice, the only moment sin the current code base when `Run()` (aka no arguments provided, including no `switched_to_mempurge`) is called is in `flush_job_test.cc`, so I am (arguably)  hesitant to make it a required argument for this reason.

Reply:
Make sense. Thanks for clarifying it.